lib/rate-limit.js

```diff
@@ -1,39 +1,47 @@
-export default function rateLimit (instance, maxRetry = 5) {
-  let attempt = 0
-  let lastErrorAt
+const attempts = {}
 
+export default function rateLimit (instance, maxRetry = 5) {
   instance.interceptors.response.use(function (response) {
     // we don't need to do anything here
     return response
   }, function (error) {
-    const {config} = error
-    if (!config || !instance.defaults.retryOnError) {
+    const {response, config} = error
+    if (!response || !config || !instance.defaults.retryOnError) {
       return Promise.reject(error)
     }
-    // in axios reponse will be available only if there a response from server
-    // in this case will assume it is a 500 error
-    error.response = error.response || {status: 500}
-    if (error.response.status === 500) {
-      attempt++
-      lastErrorAt = lastErrorAt || Date.now()
-      lastErrorAt = Date.now()
-      // we reject if there are too much errors in a short period of time
-      if (attempt >= maxRetry) {
+
+    let retryErrorType = null
+    let wait = 0
+
+    if (!response) {
```

Comment:
@Khaledgarbaya I reverted the last commit again. The check was fine, I straight want to reject when response and request are missing, but when there is a request but no response, we got a connection error.

We may replace this line with:
```
if (request && !response) {
```